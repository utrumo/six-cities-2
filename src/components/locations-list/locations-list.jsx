import React from 'react';
import PropTypes from 'prop-types';
import Location from '../location/location.jsx';

import {connect} from 'react-redux';
import {getLocations, getCurrentLocation} from '../../store/data/selectors.js';
import {ActionCreator} from '../../store/data/data.js';

const LocationsList = ({locations, currentLocation, onClick}) => (
  <section className="locations container">
    <ul className="locations__list tabs__list">
      {locations.map((it, i) => (
        <Location
          key={i}
          location={it}
          current={it === currentLocation}
          onClick={() => onClick(it)}
        />
      ))}
    </ul>
  </section>
);

LocationsList.propTypes = {
  locations: PropTypes.arrayOf(PropTypes.string),
  currentLocation: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  locations: getLocations(state),
  currentLocation: getCurrentLocation(state),
});

const mapDispatchToProps = {
  onClick: ActionCreator.changeLocation,
};

export {LocationsList};
export default connect(mapStateToProps, mapDispatchToProps)(LocationsList);
