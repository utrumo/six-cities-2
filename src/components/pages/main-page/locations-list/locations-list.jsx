import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {DataSelector, DataOperation} from 'store';
import Location from '../location/location.jsx';

const LocationsList = ({locations, currentLocation, onClick}) => (
  <section className="locations container">
    <ul className="locations__list tabs__list">
      {locations.map((it, i) => (
        <Location
          key={i}
          location={it}
          current={it === currentLocation}
          onClick={() => onClick(it)}
        />
      ))}
    </ul>
  </section>
);

LocationsList.propTypes = {
  locations: PropTypes.arrayOf(PropTypes.string),
  currentLocation: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  locations: DataSelector.getLocations(state),
  currentLocation: DataSelector.getCurrentLocation(state),
});

const mapDispatchToProps = {
  onClick: DataOperation.changeLocation,
};

export {LocationsList};
export default connect(mapStateToProps, mapDispatchToProps)(LocationsList);
