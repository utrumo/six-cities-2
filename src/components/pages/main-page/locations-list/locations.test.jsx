import React from 'react';
import renderer from 'react-test-renderer';
import {LocationsList} from './locations-list.jsx';

it(`LocationsList should renderes correctly`, () => {
  const locations = [`Moscow`, `New York`];
  const tree = renderer
    .create(<LocationsList locations={locations} currentLocation="Moscow" onClick={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
