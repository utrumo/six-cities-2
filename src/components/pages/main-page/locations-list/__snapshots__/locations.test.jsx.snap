// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationsList should renderes correctly 1`] = `
<section
  className="locations container"
>
  <ul
    className="locations__list tabs__list"
  >
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item tabs__item--active"
        href={null}
        onClick={[Function]}
        style={
          Object {
            "cursor": "default",
          }
        }
      >
        <span>
          Moscow
        </span>
      </a>
    </li>
    <li
      className="locations__item"
    >
      <a
        className="locations__item-link tabs__item"
        href="#"
        onClick={[Function]}
        style={null}
      >
        <span>
          New York
        </span>
      </a>
    </li>
  </ul>
</section>
`;
